ci/cirrus_general_ci.yml

```diff
@@ -56,88 +62,86 @@ linux_aarch64_test_task:
     
-    _CWD=$PWD
-    echo "_CWD=$(_CWD)" >> $CIRRUS_ENV
-    cd $CIRRUS_WORKING_DIR/..
-    rm -rf $CIRRUS_WORKING_DIR
-
-  pip_cache:
-    folder: ~/.cache/pip
-
-  <<: *MODIFIED_CLONE
-
-  python_dependencies_script: |
-    cd $_CWD
-    python -m pip install cython
-    pip install --upgrade --pre -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple numpy
-    python -m pip install meson ninja pybind11 pythran pytest
-    python -m pip install click rich_click doit pydevtool pooch
+    # this line tells us where the task keeps its pip cache. 
+    # It may change over time!
+    echo $(python -m pip cache dir)
+    echo $(python -m pip cache list)
 
   build_script: |
-    python dev.py build
+    python dev.py build -j1
 
   test_script: |
-    set -xe -o
-    python dev.py test
+    python dev.py test -j1
+    ccache -s
 
 
 macos_arm64_test_task:
   macos_instance:
-    image: ghcr.io/cirruslabs/macos-monterey-xcode:14
+    image: ghcr.io/cirruslabs/macos-monterey-xcode:latest
 
   <<: *MODIFIED_CLONE
 
+  #  ccache_cache:
+  #    folder: .ccache
+  #    populate_script:
+  #      - mkdir -p .ccache
+  #    fingerprint_key: ccache-macosx_arm64
+
   pip_cache:
-    folder: ~/.cache/pip
+    folder: /Users/admin/Library/Caches/pip
 
-  test_script: |
+  prepare_env_script: |
     brew install python@3.10
 
     export PATH=/opt/homebrew/opt/python@3.10/libexec/bin:$PATH
+    export PYTHONPATH=/opt/homebrew/lib/python3.10/site-packages
+    
+    echo "PYTHONPATH=$PYTHONPATH" >> $CIRRUS_ENV
+    echo "_CWD=$PWD" >> $CIRRUS_ENV
+    echo "PATH=$PATH" >> $CIRRUS_ENV
+    #    export PATH=/opt/homebrew/opt/ccache/libexec:$PATH
+
     python --version
 
     # used for installing OpenBLAS/gfortran
     bash tools/wheels/cibw_before_build_macos.sh $PWD
+    echo "DYLD_LIBRARY_PATH=/usr/local/gfortran/lib:/opt/arm64-builds/lib" >> $CIRRUS_ENV
 
-    export PKG_CONFIG_PATH=/opt/arm64-builds/lib/pkgconfig
-    export CMAKE_PREFIX_PATH=/opt/arm64-builds/
-
-    pushd ~/
-    python -m venv scipy-dev
-    source scipy-dev/bin/activate
-    popd
+    echo "PKG_CONFIG_PATH=/opt/arm64-builds/lib/pkgconfig" >> $CIRRUS_ENV
+    echo "CMAKE_PREFIX_PATH=/opt/arm64-builds/" >> $CIRRUS_ENV
+    # export CCACHE_DIR=$PWD/.ccache
+    
+    cd ..
+    python -m venv test_env
+    source test_env/bin/activate
+    cd $_CWD
 
     python -m pip install meson ninja numpy cython pybind11 pythran cython
     python -m pip install click rich_click doit pydevtool
-    python -m pip install pytest pooch
-    export DYLD_LIBRARY_PATH=/usr/local/gfortran/lib:/opt/arm64-builds/lib
-    python dev.py test
+    python -m pip install pytest pooch pytest-xdist
+    
+    # this line tells us where the task keeps its pip cache. 
+    # It may change over time!
+    echo $(python -m pip cache dir)
+    echo $(python -m pip cache list)
+
+  build_script: |
+    cd ..
+    source test_env/bin/activate
+    cd $_CWD
+    python dev.py build -j3
```

Comment:
Is the `-j3` a leftover here from before splitting build/test? Either leaving it out or using `-j4` should be faster. This is `ninja`'s parallelism, which scales way better than `pytest-xdist`.

Reply:
I can take it out. I was trying to optimize build/test usage of CPU, and doing this by experimentation rather than knowing things.